### PR TITLE
Numb

### PR DESCRIPTION
**Pull request by zach1502**
cuz why not,
ill use the many skills i have acquired by making squid pie to combine this with guess later, just need to check if it works.

### Type
- [x] No U
- [ ] Bugfix
- [ ] Enhancement
- [ ] New feature
- [x] No I

### CODE REVIEW

File: cogs/numb.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+'''This is a guess the number game'''
+import random
+import asyncio
+from discord.ext import commands
+
+
+GuessesTaken = 0
+
+class Guess:
+	'''this is a number guessing game'''
```
Comment (Injabie3) [inline cogs/numb.py]: Let's be consistent: 4 spaces per indentation block, please.  Here, and throughout.

File: cogs/numb.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+		else:
+			await self.bot.say('type again')
+			return
+
+		await self.bot.say('I am thinking of a number between 1 and 100.')
+		while guessesTaken != 5:
+
+  			await self.bot.say('Take a guess.')
+  			guess = self.bot.wait_for_message(channel=context.message.channel, timeout = 30)
```
Comment (Injabie3) [inline cogs/numb.py]: nit: PEP8:  `timeout = 30)` -> `timeout=30)`

File: cogs/numb.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  			if not guess:
+  				await self.bot.say('you think too slowly, imma just leave now')
```
Comment (Injabie3) [inline cogs/numb.py]: nit: Proper grammar, please.

File: cogs/numb.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  				return
+  			
+  			guess = int(guess)
```
Comment (Injabie3) [inline cogs/numb.py]: Need to handle case where you cannot cast to an `int`.

File: cogs/numb.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  			guessesTaken = guessesTaken + 1
+
+  			if guess < number:
+    			    print('Your guess is too low.')
```
Comment (Injabie3) [inline cogs/numb.py]: You actually want to output this to the user, so you want the following instead here (and in the other prints below):
```py
await self.bot.say(...)
```

File: cogs/numb.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    		
+  			if guess == number:
+    			    break
+    		
+		if guess == number:
+			guessesTaken = str(guessesTaken)
+			print('Good job! You guessed my number in ' + guessesTaken + ' guesses!')
+		if guess != number:
+  			number = str(number)
+  			print('Nope. The number I was thinking of was ' + number)
```
Comment (Injabie3) [inline cogs/numb.py]: Missing `def setup(bot):` function.

File: cogs/numb.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
+'''This is a guess the number game'''
+import random
+import asyncio
+from discord.ext import commands
+
+
+GuessesTaken = 0
+
+class Guess:
```
Comment (Injabie3) [inline cogs/numb.py]: This conflicts with #74.  Please choose a different name, or abandon your previous PR.

File: cogs/numb.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
+'''This is a guess the number game'''
+import random
+import asyncio
+from discord.ext import commands
+
+
+GuessesTaken = 0
```
Comment (Injabie3) [inline cogs/numb.py]: This seems like a local variable.  Please keep it internal to your class, or better yet, your `numguess` method.

File: cogs/numb.py [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+class Guess:
+	'''this is a number guessing game'''
+	def __init__(self, bot):
+		self.bot = bot
+	@commands.command(name="numguess", pass_context=True)
+	async def mycom(self, context):
+		'''this will start a number guessing game'''
+
+		await self.bot.reply("how hard would you like the number game to be? easy/med/hard")
```
Comment (Injabie3) [inline cogs/numb.py]: `await self.bot.say(...)`

File: cogs/numb.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+			number = random.randint(1,100)
+
+		#easter egg
+		elif mode == 'no u':
+			number = randome.randint(1,420)
+			await self.bot.say('no u')
+
+		else:
+			await self.bot.say('type again')
+			return
```
Comment (Injabie3) [inline cogs/numb.py]: FYI, the user doesn't get a chance to retype.  They have to reinvoke the original command again.

**Review by Injabie3 — CHANGES_REQUESTED** [review]
Hey zach, please see comments below.

File: cogs/numb.py (hunk 10 above)
Comment (zach1502) [reply cogs/numb.py]: shhhhh.....
no one needs to know.......

File: cogs/numb.py [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+	    elif mode == 'hard':
+		number = random.randint(1,100)
+
+	    #easter egg
+	    elif mode == 'no u':
+		number = random.randint(1,420)
+		await self.bot.say('no u')
+	    else:
+		await self.bot.say('type again')
+                while True:
```
Comment (KamiCreed) [inline cogs/numb.py]: Wouldn't this cause the bot to continuously ask to type again even if it timeouts?

File: cogs/numb.py [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+  		guess = self.bot.wait_for_message(channel=context.message.channel, timeout=30)
+  		if not guess:
+  		    await self.bot.say('you think too slowly, I am going to leave now...')
+  		    return
+
+  		while True:
+		    try:
+			guess = int(guess)
+		    except ValueError:
+			await self.bot.say('That\'s no a number, type in a number please')
```
Comment (KamiCreed) [inline cogs/numb.py]: no**t** a number?

**Review by KamiCreed — CHANGES_REQUESTED** [review]

**Comment by KamiCreed** [comment]
What are these commit messages... They don't really tell us very much and is pretty ugly.... [Please read this.](https://chris.beams.io/posts/git-commit/)

File: cogs/numb.py [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+            if guess > number:
+                await self.bot.say('Your guess is too high.')
+
+            if guess == number:
+                return
+
+        if guess == number:
+            await self.bot.say('Good job! You guessed my number in ' + str(guessesTaken) + ' guesses!') # pylint: disable=line-too-long
+
+        if guess != number:
```
Comment (Injabie3) [inline cogs/numb.py:83]: If you said `guess == number` above, then just make this an else.

File: cogs/numb.py [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+        else:
+            await self.bot.say('type again')
+            mode = self.bot.wait_for_message(channel=context.message.channel, timeout=30)
+
+        while True:
+            try:
+                if mode not in ["easy", "mid", "hard", "no u"]:
+                    raise ValueError
+            except ValueError:
+                await self.bot.say('type again')
```
Comment (Injabie3) [inline cogs/numb.py:44]: This code is being repeated here and above, and is difficult to follow.  Can you find a way to structure your "checking mode" logic to loop a few times before giving up?  Ultimately, we don't want infinite loops in a command.

File: cogs/numb.py [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+                try:
+                    guess = int(guess)
+                except ValueError:
+                    await self.bot.say("That's not a number, type in a number please")
+                    guess = self.bot.wait_for_message(channel=context.message.channel, timeout=30)
+                    continue
+                else:
+                    break
+
+            guessesTaken += guessesTaken
```
Comment (Injabie3) [inline cogs/numb.py:69]: *2, or just +1?

File: cogs/numb.py [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+
+            if guess < number:
+                await self.bot.say('Your guess is too low.')
+
+            if guess > number:
+                await self.bot.say('Your guess is too high.')
+
+            if guess == number:
+                return
```
Comment (Injabie3) [inline cogs/numb.py:78]: This will exit your command without printing that your guess was correct.

File: cogs/numb.py [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+
+            if not guess:
+                await self.bot.say('you think too despacito, I am going to leave now...')
+                return
+
+            while True:
+                try:
+                    guess = int(guess)
+                except ValueError:
+                    await self.bot.say("That's not a number, type in a number please")
```
Comment (Injabie3) [inline cogs/numb.py:63]: Similarly here, can we structure this so that it is not an infinite loop?

**Review by Injabie3 — CHANGES_REQUESTED** [review]
Hey zach, here's some feedback.

**Comment by Injabie3** [comment]
Closing as no commits within the last month.

**Pull request closed without merging**